### ISSUE TITLE
Unable to follow some PeerTube instances from Mastodon

### ISSUE THREAD

**Issue by FediVideos** [tootsuite/mastodon#15347]
I previously raised this issue on the PeerTube git but they could not find the source of the problem, even when looking through error logs from affected PeerTube instance. That's why I am now raising it here.

Masto-PT follows were working fine a few months ago, perhaps it might be due to something that happened with a Mastodon update since then?

Existing follows are unaffected, this bug only seems to affect new following attempts.

### Expected behaviour

Clicking on PeerTube account "Follow" button within Mastodon should follow the account, it should appear in the Mastodon account's "following" list.

### Actual behaviour

For some instances the follow works fine.

For other instances it initially seems to work, but when you refresh the PeerTube account profile page via Mastodon the follow has been turned into a follow request. Also, the PeerTube account does not appear in the Mastodon "following" list, and its new posts do not appear in the timeline.

### Steps to reproduce the problem

1. Log into your Mastodon account
2. Search for these accounts and click on "Follow" for each of them: @pine64tilvids@tilvids.com, @blitzcitydiy@diode.zone and @documentaries@vidcommons.org (NOTE: Make sure you are not already following these before you attempt to reproduce the problem, because pre-existing follows made months ago are unaffected)
3. Check your Mastodon "Following" list, refresh it to make sure it is up to date. Check if you can see all three of these accounts added to your following list. One or more of these accounts will be missing from the list.
4. Within Mastodon, refresh the profile pages for all three of these accounts. You should see the "Unfollow" button turn into a "Cancel follow request" button for one or more of these accounts.

### Specifications

Mastodon v3.3.0rc1 - Bug observed from three different Mastodon instances (mas.to, mastodon.online and mstdn.social) all running v3.3.0rc1

PeerTube 2.4.0 - Bug observed when following accounts from some PT instances, but other instances work fine, even when all instances are running 2.4.0

Browsing was on Firefox 83.0 on Ubuntu 20.04

![MastoNoFollow3](https://user-images.githubusercontent.com/75383929/102125025-eeca5a80-3e40-11eb-98f6-ec3cb1bcc767.png)

**Comment by Gargron** [comment]
> For other instances it initially seems to work, but when you refresh the PeerTube account profile page via Mastodon the follow has been turned into a follow request. Also, the PeerTube account does not appear in the Mastodon "following" list, and its new posts do not appear in the timeline.

All follows are follow requests which the remote server has to accept. The difference with a "locked" account in Mastodon is simply that the accept is manual and not automatic. So if there is a problem with the PeerTube server not accepting follow requests, this is what you would observe...

**Comment by FediVideos** [comment]
I messaged the owner of Tilvids.com when this started happening, they hadn't altered any settings or done any updates at their end. They emailed their logs to chocobozzz (the PT lead dev) but chocobozzz couldn't find anything wrong, so that's what brought me here.

Even weirder, Vidcommons.org used to work fine, then this bug appeared, then the bug went away again. I messaged the owner of Vidcommons.org too when the bug appeared and then again when it disappeared. They hadn't done anything new, hadn't made any alterations.

It's very weird.
